test/system/170-run-userns.bats

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bats   -*- bats -*-
+# shellcheck disable=SC2096
+#
+# Tests for podman build
+#
+
+load helpers
+
+@test "podman --group-add keep-groups while in a userns" {
+    skip_if_rootless "choot is not allowed in rootless mode"
+    run chroot --groups 1234 / ${PODMAN} run --uidmap 0:200000:5000 --group-add keep-groups $IMAGE id
+    is "$output" ".*65534(nobody)" "Check group leaked into user namespace"
+}
+
+@test "podman --group-add keep-groups while not in a userns" {
+    skip_if_rootless "choot is not allowed in rootless mode"
+    skip_if_remote "--group-add keep-groups not supported in remote mode"
+    run chroot --groups 1234,5678 / ${PODMAN} run --group-add keep-groups $IMAGE id
+    is "$output" ".*1234" "Check group leaked into container"
+}
+
+@test "podman --group-add without keep-groups while in a userns" {
+    skip_if_rootless "choot is not allowed in rootless mode"
+    skip_if_remote "--group-add keep-groups not supported in remote mode"
+    run chroot --groups 1234,5678 / ${PODMAN} run --uidmap 0:200000:5000 --group-add 457 $IMAGE id
+    is "$output" ".*457" "Check group leaked into container"
+}
+
+@test "podman --group-add with keep-groups {
+    if is_remote; then
+        run_podman run 125 --group-add keep-groups $IMAGE id
+        is "$output" ".*keep-groups not supported in remote mode" "Check keep-groups in remote mode"
+    else
+        run_podman run --group-add keep-groups $IMAGE id
+        is "$output" "uid=0(root) gid=0(root) groups=0(root)" "Check group leaked while root"
+    fi
+}
+
+@test "podman --group-add without keep-groups {
+    run_podman run --group-add 457 $IMAGE id
+    if is_rootless; then
+        is "$output" ".*65534(nobody)" "Check group leaked into rootless mode"
+    else
+       is "$output" ".*457" "Check group leaked into container"
+    fi
+}
+
+@test "podman --group-add keep-groups plus added groups {
+    run_podman 125 run --group-add keep-groups --group-add 457 $IMAGE id
+    is "$output" ".*the '--group-add keep-groups' option is not allowed with any other --group-add options" "Check group leaked into container"
+}
+
+function teardown() {
+    # A timeout or other error in 'build' can leave behind stale images
+    # that podman can't even see and which will cascade into subsequent
+    # test failures. Try a last-ditch force-rm in cleanup, ignoring errors.
+    run_podman '?' rm -a -f
+    run_podman '?' rmi -f build_test
+
+    basic_teardown
+}
```

Comment:
As long as you're resubmitting, might as well remove this entire `teardown`; it's not necessary here